### PR TITLE
[CBRD-21654] server side loaddb

### PR DESCRIPTION
**Pull request by valerinistor**
http://jira.cubrid.org/browse/CBRD-21654

Add loaddb implementation on server which includes parsing and inserting of data rows.
This is a preliminary version and pull request is labeled with `in progress`. Pull request was open with the idea to start collecting feedback, please take into account TODOs

Changes done regarding source files structure:

* Renamings
  * src/loaddb/driver.cpp -> src/loaddb/load_driver.cpp
  * src/loaddb/driver.hpp -> src/loaddb/load_driver.hpp
  * src/loaddb/scanner.hpp -> src/loaddb/load_scanner.hpp
  * src/executables/loaddb.c -> src/loaddb/load_db.c
  * src/executables/loader.c -> src/loaddb/load_sa_loader.c
  * src/executables/loader.h -> src/loaddb/load_sa_loader.h
  * src/executables/loader_grammar.y -> src/loaddb/load_grammar.yy
  * src/executables/loader_lexer.l -> src/loaddb/load_lexer.l
  * src/executables/loader_object_table.c -> src/loaddb/load_object_table.c
  * src/executables/loader_object_table.h -> src/loaddb/load_object_table.h
  * src/executables/load_object.c -> src/loaddb/load_object.c
  * src/executables/load_object.h -> src/loaddb/load_object.h

* Removed (not used)
  * src/executables/loader_disk.c
  * src/executables/loader_disk.h

* New files added
  * src/loaddb/load_common.cpp
  * src/loaddb/load_common.hpp
  * src/loaddb/load_db_value_converter.cpp
  * src/loaddb/load_db_value_converter.hpp
  * src/loaddb/load_manager.cpp
  * src/loaddb/load_manager.hpp
  * src/loaddb/load_server_loader.cpp
  * src/loaddb/load_server_loader.hpp

### CODE REVIEW

File: src/loaddb/manager.hpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  static const std::size_t DRIVER_POOL_SIZE = 1;
+
+  // TODO CBRD-21654 add class documentation
+  class manager
+  {
+    public:
+      manager (const manager &copy) = delete;
+      manager &operator= (const manager &other) = delete;
+      manager &operator= (const manager &&other) = delete;
```
Comment (paulgheorghecristian) [inline src/loaddb/manager.hpp]: here, for move operator, const doesn't make sense

Comment (valerinistor) [reply src/loaddb/manager.hpp]: you are right ;)

File: src/storage/heap_file.h [hunk 2]
```diff
@@ -380,7 +380,7 @@ extern int heap_scancache_quick_start (HEAP_SCANCACHE * scan_cache);
 extern int heap_scancache_quick_start_modify (HEAP_SCANCACHE * scan_cache);
 extern int heap_scancache_end (THREAD_ENTRY * thread_p, HEAP_SCANCACHE * scan_cache);
 extern int heap_scancache_end_when_scan_will_resume (THREAD_ENTRY * thread_p, HEAP_SCANCACHE * scan_cache);
-extern void heap_scancache_end_modify (THREAD_ENTRY * thread_p, HEAP_SCANCACHE * scan_cache);
+extern int heap_scancache_end_modify (THREAD_ENTRY * thread_p, HEAP_SCANCACHE * scan_cache);
```
Comment (ribram) [inline src/storage/heap_file.h]: Remove code. It is almost never checked. I'd rather remove the code from heap_scancache_force_modify and heap_scancache_quick_end.

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/loaddb/load_client_loader.c (1113 incorrect indentation(s) found)
```
    73 const
    74   std::size_t
    75   LDR_MAX_ARGS = 32;
   175 typedef struct ldr_context
   176   LDR_CONTEXT;
   197   DB_ATTDESC *
   198     attdesc;			/* Attribute descriptor */
   199   SM_ATTRIBUTE *
   200     att;			/* Attribute */
   201   LDR_SETTER
   202     setter[NUM_LDR_TYPES];	/* Setter functions indexed by type */
   203 
   204   char *
   205     parser_str;			/* used as a holder to hold the parser strings when parsing method arguments. */
   206   size_t
   207     parser_str_len;		/* Length of parser token string */
   208   size_t
   209     parser_buf_len;		/* Length of parser token buffer */
   210   data_type
   211     parser_type;		/* Used when parsing method arguments, to store */
```
... More than 20 lines are omitted.
#### File: src/loaddb/load_db.c (15 incorrect indentation(s) found)
```
   754      bool load_on_server = ldr_load_on_server ();
   755      if (load_on_server)
   756      {
   757      char object_file_abs_path[PATH_MAX];
   758      if (realpath (args.object_file, object_file_abs_path) != NULL)
   759      {
   760      loaddb_load_object_file (object_file_abs_path);
   761      }
   762      // in fact there is a longer story here
   763      }
   764      else
   765      {
   766      // probably some error
   767      }
   768    */
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Pull request merged**